### PR TITLE
Jech master merge

### PR DESCRIPTION
**Pull request by drozdziak1**
This PR merges upstream babeld's `master` and introduces small changes to the tests.

I'm pushing this into the `althea` branch because there's still issues like the [messy logging](https://github.com/althea-mesh/babeld/issues/5), occasional counter-intuitive variable names or the incorrectly parsed packets @jkilpatr mentioned [here](https://github.com/althea-mesh/babeld/issues/6)

# Viewing advice
It might just be easier to view this if you pull `jech-master-merge` and just take a look at the merge commit WRT `althea` and `jech/master` with a proper difftool.

### CODE REVIEW

**Comment by drozdziak1** [comment]
Note: The cherry-pick from `travis-tests` makes #7 obsolete.

File: configuration.c [hunk 1]
```diff
@@ -339,6 +339,17 @@ get_interface_type(int c, int *type_r, gnc_t gnc, void *closure)
     return c;
 }
 
+static void
```
Comment (jkilpatr) [inline configuration.c:342]: what does this do? out of curiosity.

**Review by jkilpatr — COMMENTED** [review]
Lots of interesting changes here, I might want to pick your mind on some of what babel did but your integration of the changes is neat. We probably want to wait for upstream to finish, add one more commit to this tree then land it. 

Sound good?

File: configuration.c (hunk 1 above)
Comment (drozdziak1) [reply configuration.c:342]: It's not mine, I believe it's just some helper that Juliusz added. Honestly this merge was pretty straightforward with maybe one or two gotchas and only rebasing was really causing any trouble

**Comment by drozdziak1** [comment]
Sure, Our changes still need to integrate better with the project (think docs for our CLI options and config values, polishing variable names and debug calls) so I'll probably tend to that in the meantime.

**Comment by drozdziak1** [comment]
This PR is superseded by #10

**Pull request closed without merging**